node/src/Pos/Client/Txp/Util.hs

```diff
@@ -74,19 +74,27 @@ data TxRaw = TxRaw
     -- ^ Remaining money
     }
 
-data TxError
-    = TxError { unTxError :: !Text }
+data TxError =
+      NotEnoughMoney !Coin    -- ^ Parameter: how much more money is needed
+    | FailedToStabilize !Int  -- ^ Parameter: how many attempts were performed
+    | OutputIsRedeem !Address
+    | RedemptionDepleted
+    | GeneralTxError !Text    -- ^ Parameter: description of the problem
     deriving (Show, Generic)
 
 instance Exception TxError
 
 instance Buildable TxError where
-    build (TxError msg) = bprint ("Transaction creation error ("%stext%")") msg
-
-throwTxError
-    :: MonadError TxError m
-    => Text -> m a
-throwTxError = throwError . TxError
+    build (NotEnoughMoney coin) =
+        bprint ("Transaction creation error: not enough money, need "%build%" more") coin
+    build (FailedToStabilize iters) =
+        bprint ("Transaction creation error: failed to stabilize fee after "%build%" iterations") iters
+    build (OutputIsRedeem addr) =
+        bprint ("Destination address "%build%" is a redeem address") addr
+    build RedemptionDepleted =
+        bprint "Redeem balance is 0"
```

Comment:
Eh, what does it meen? "Redeem address balance" or what?

Reply:
Fixed.